audobject/core/resolver.py

```diff
@@ -179,6 +179,27 @@ class Function(Base):
     Encodes source code of function and
     dynamically evaluates it when the value is decoded again.
 
+    Note that a decoded function
+    might raise a :class:`NameError`,
+    if it relies on objects or functions
+    that are not defined or imported
+    inside the function.
+    For instance,
+    in this example
+    function ``plus_1()`` requires
+    that ``_plus_1()`` is locally defined:
```

Comment:
```suggestion
    For instance,
    the following example will raise an error
    since ``plus_1()`` relies on ``_plus_1()``,
    which is defined outside the function:
```

Reply:
No, the example does not raise an error. To get an error we would need to change the example to:

```python
   .. code-block:: python

        def _plus_1(x):
            return x + 1

        def plus_1(x):
            return _plus_1(x)  # calls local function -> not serializable

        resolver = Function()
        encoded_value = resolver.encode(plus_1)
        del _plus_1
        decoded_value = resolver.decode(encoded_value)
        decoded_value(1)
```

Would you prefer to have this more explicit example?

Reply:
Yes, or maybe we change the text to:

```    
For instance,
here ``plus_1()`` is not serializable
since it relies on ``_plus_1()``,
which is defined outside the function:
```